### PR TITLE
Finished AdaBoost classifier

### PR DESCRIPTION
**Pull request by beelze-b**
AUROC of .867 on test set

Highest weighted 3 were: KLRG2, METTL7B, and GPR158

Using a decision stump classifier as the weak learner for AdaBoost, the best parameters were:
a learning rate of  0.5 and 50 estimators.

More estimators seem to perform best, and more hyperparameter tuning can be done.

### CODE REVIEW

**Comment by dhimmel** [comment]
Nice pull request -- seeing these results is helpful.

Can you rename `AdaBoost_NabeelSarwar.ipynb` to `AdaBoost-NabeelSarwar.ipynb`?

Also can you export your notebook to a .py file for commenting? You can do this by running the following command from the `algorithms` directory:

``` sh
jupyter nbconvert --to=script --FilesWriter.build_directory=scripts AdaBoost-NabeelSarwar.ipynb
```

Cell 24 image has axis labels reversed.

**Comment by yl565** [comment]
A few suggestions:
1. Use `n_feature_kept` > 5000
2. Try grid search `max_depth` and `min_samples_split` of `DecisionTreeClassifier` which affects overfitting/underfitting. Keep `learning_rate` and `n_estimators` fixed if computational time too long

**Comment by beelze-b** [comment]
Hi, 

I actually realized I used decision trees instead of decision stumps as is traditional with AdaBoost. This was corrected.

AUROC of .905 on test set now. 
Same best parameters.
Highest 3 weighted features are now:
1. SPATA18
2. EDA2R
3. CDKN2A

I followed the corrections that Daniel made.

I followed Yichuan's first suggestion.
For the second one, I don't see a priori why I would use decision trees instead of stumps. Decision trees are more typical of bagging models from what I understand. I am not saying that performance wouldn't increase, but from my experience, AdaBoost benefits from the low variance of the weak learners.

Also, the computational time wasn't too long. It was only around 10 minutes on my modest machine.

**Comment by yl565** [comment]
I'm just curious if there is certain bias/variance trade off for this data set. Although I haven't encountered a example myself, I remember I read somewhere that if the weak learners are too week it would negatively affect performance.  In any case, thanks for the PR.

**Comment by dhimmel** [comment]
@yl565 are any of your comments on this PR outstanding or are we good to merge?

**Comment by yl565** [comment]
Merged
